### DISCUSSION TITLE
localePath or localeStructure does not work as expected

### DISCUSSION THREAD

**Discussion by ivoba** [i18next/next-i18next#2073] [Q&A]
Hi, 
i am having a hard time figuring out how to change the path and filename of the translation file.
I tried both options described in the docs but both do not seem to work.
Desired output is: `public/locales/{namespace}.{locale}.json`

I am using
```
    "next": "13.1.1",
    "next-i18next": "^13.0.2",
    "react-i18next": "^12.1.1",
```

**localeStructure**
So i left localPath on default and changed localeStructure
i18next-parser.config.js:

```
module.exports = {
  i18n: {
    locales: ['de', 'en'],
    defaultLocale: 'de',
    localeStructure: '{{ns}}.{{lng}}'
  }
}
```
This works only if i also keep files in the default dirs like:
`public/locales/de/common.json`
then it would pick up values from:
`public/locales/common.de.json`

If only having the latter it says: `i18next::translator: missingKey de common header.logo.claim header.logo.claim`

But files should not be double, this feels rather wrong.

**localePath**
I added a function as localePath adn i receive other errors;

```
module.exports = {
  i18n: {t
    locales: ['de', 'en'],
    defaultLocale: 'de',
    localePath: (locale, namespace) => (`./public/locales/${namespace}.${locale}.json`),
  },
}
```
Yields this error:

> Error: Must provide all namespaces in ns option if using a function as localePath

Adding a namespace list manually somewhere feels also rather wrong.

And even if i add `ns: ['common'],` i get another error:

> Error: Error serializing `._nextI18Next.userConfig.i18n.localePath` returned from `getStaticProps` in "/".
Reason: `function` cannot be serialized as JSON. Please only return JSON serializable data types.

What am i missing here?

**Comment by ivoba** [comment]
So at least i got the serialization working now with following the docs with `serializeConfig: false,` 
https://github.com/i18next/next-i18next#unserializable-configs

> export default appWithTranslation(App, nextI18NextConfig);

But still having to add all namespaces manually still seems to be cumbersome.

**Comment by adrai** [comment]
Having a custom localeStructure that is not listable requires also to have the ns option defined. v13.0.3 will error now if not done so.
With the default localeStructure next-i18next tries to automatically preload all namespaces of the appropriate language... basically `ls {{lng}}`
But with a localeStructure like yours this is not possible.
